crypto/salsa20/stream_salsa20.c

```diff
@@ -66,6 +66,11 @@ _crypto_stream_salsa20_pick_best_implementation(void)
 {
   if(implementation)
     return 0;
+#if defined(iOS)
```

Comment:
What's the point of this?

If AVX2 and SSE2 aren't available, which they aren't on iOS hardware, this implementation will be chosen anyway (see lines [64](https://github.com/loki-project/loki-network/blob/35cc9fcb52bc6cfe0cb3ec3ca058e9ac7a05f920/crypto/salsa20/stream_salsa20.c#L64)-[91](https://github.com/loki-project/loki-network/blob/35cc9fcb52bc6cfe0cb3ec3ca058e9ac7a05f920/crypto/salsa20/stream_salsa20.c#L91))

If this was to fix a compile time error, you were probably trying to build for the wrong platform.

Reply:
does the iPhone simulator not exist? last time I checked, its target is `[i586|x86_64]-apple-iphoneos*` or similar